### DISCUSSION TITLE
finite volume - output values with coordinates

### DISCUSSION THREAD

**Discussion by aiskhak** [idaholab/moose#19986] [Q&A Modules: Navier-Stokes] [answered]
Hello All,

I am using INSFV kernels for simulations. Then I open exodus file with the results and want to save the data in csv file. However, all the values are stored as "cells" with NO coordinates. I need somehow save the data with coordinates. How do I do this?

Thanks!

**Comment by GiudGiud** [comment, marked as answer]
Hello

You could output each coordinate into a different variable using a ParsedAux with just ‘x’ (respectively y and z) for the function. 

You ll need to have the use_xyt parameter set to `true`.

Alternatively, have you tried using this vector postprocessor to get the variable values directly to CSV ?
https://mooseframework.inl.gov/source/vectorpostprocessors/ElementValueSampler.html 

Guillaume

**Reply by aiskhak** [reply]
Thank you, ElementValueSampler is exactly what I need.
